scripts/deployMainnet.ts

```diff
@@ -4,28 +4,21 @@
  * In such setup xchain messages doesn't work.
  */
 import hre from 'hardhat'
-import {
-  AuthLike,
-  Dai,
-  L1ERC20Gateway,
-  L1Escrow,
-  L1GovernanceRelay,
-  L2DepositedToken,
-  L2GovernanceRelay,
-} from '../typechain'
-import { ZERO_GAS_OPTS, deployContract, getL2Factory, MAX_UINT256 } from '../test-e2e/helpers/utils'
 import { JsonRpcProvider } from '@ethersproject/providers'
+import { deploy } from './common'
+import { ZERO_GAS_OPTS } from '../test-e2e/helpers/utils'
 
 const L1_PAUSE_PROXY_ADDRESS = '0xBE8E3e3618f7474F8cB1d074A26afFef007E98FB'
 const L1_ESM_ADDRESS = '0x29CfBd381043D00a98fD9904a431015Fef07af2f'
 const L1_DAI_ADDRESS = '0x6B175474E89094C44Da98b954EedeAC495271d0F'
 const L1_DEPLOYER_ADDRESS = '0xAb5801a7D398351b8bE11C439e05C5B3259aeC9B'
+const L2_XDOMAIN_MESSENGER = '0x4200000000000000000000000000000000000007'
 const L2_DEPLOYER_PRIV_KEY = '6ea93392eb84fad521111def2e8fbd9c45c2b085907797e1a60a210d5bf7089d'
```

Comment:
I think we should avoid putting private keys in the code. It's a bad practise as even if there is no funds on there currently, someone may forget in the future that this is public. Perhaps some gitignore'd json wallet file is a better way to go about this? Alternatively an environment variable could work.